bitnami/redis/templates/configmap.yaml

```
@@ -51,3 +51,69 @@ data:
 {{- tpl .Values.sentinel.configmap . | nindent 4 }}
 {{- end }}
 {{- end }}
+---
+{{- if .Values.sentinel.enabled }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "redis.fullname" . }}-shutdown-script
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "redis.name" . }}
+    chart: {{ template "redis.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+data:
+  shutdown.sh: |-
+    #!/bin/bash
+    MASTER_SET="{{ .Values.sentinel.masterSet }}"
+    REDIS_COMMAND="redis-cli -p 26379"
+
+    function get_master_ip {
+      echo $($REDIS_COMMAND sentinel get-master-addr-by-name mymaster | head -n 1)
```

Instead of using shell expansion for this, you could do the following:

```suggestion
    REDIS_CMD="redis-cli"
    REDIS_ARGS=("-p" "$REDIS_SENTINEL_PORT")

    get_master_ip() {
      echo $("$REDIS_CMD" "${REDIS_ARGS[@]}" sentinel get-master-addr-by-name mymaster | head -n 1) 
```

Also, like I included in the code snippet, we should be using the environment variable for the port number.